### PR TITLE
implemented Number#as_human

### PR DESCRIPTION
**Pull request by druzn3k**
This commit closes issue #25. Ugly but implements the functionalities.

### CODE REVIEW

**Comment by duijf** [comment]
Currently, because of the way the `@doc` statements are structured, all documentation is bound to the `as_human!` function. The non-throwing version doesn't have any docs associated with it.

Redid them for you (with the format I was experimenting with yesterday &mdash; this seems to look best):

``` elixir
  @doc ~S"""
  Formats and approximates `number` for human readability.

  `1200000000` becomes `"1.2 Billion"`. This is useful as larger numbers become
  harder to read.

  See `as_human_size` if you want to print a file size.

  You can also define you own unit-quantifier names if you want to use other
  decimal units (eg.: 1500 becomes "1.5 kilometers", 0.150 becomes
  “150 milliliters”, etc). You may define a wide range of unit quantifiers, even
  fractional ones (centi, deci, mili, etc).

  # Options

  - `:locale` (atom) --- Locale to be used for formatting. *Default:*
  current locale.
  - `:precision` (integer) --- Precision of the number. *Default:* `3`
  - `:significant` - If true, precision will be the # of significant_digits. If
  false, the # of fractional digits (defaults to true)
  - `:separator` (string) --- Separator between the fractional and integer
  digits. *Default:* `"."`
  - `:delimiter` (string) --- Thousands delimiter. *Default:* `""`
  -`:units` (keyword list/string) --- Keyword list of unit quantifier names,
  *or* a string containing an i18n scope pointing to it.
  - `:format` (string) --- Format of the output string. `%u` is the quantifier,
  `%n` is the number. *Default:* `"%n %u"`

  # i18n

  This function takes a keyword list of quantifier names to use for formatting.
  It supports the following keys:

  - `:unit`
  - `:ten`
  - `:hundred`
  - `:thousand`
  - `:million`
  - `:billion`
  - `:trillion`
  - `:quadrillion`
  - `:deci`
  - `:centi`
  - `:milli`
  - `:micro`
  - `:nano`
  - `:pico`
  - `:femto`

  # Examples

      iex> Number.as_human(123)
      "123"

      iex> Number.as_human(1234)
      "1.23 Thousand"

      iex> Number.as_human(12345)
      "12.3 Thousand"

      iex> Number.as_human(1234567)
      "1.23 Million"

      iex> Number.as_human(1234567890)
      "1.23 Billion"

      iex> Number.as_human(1234567890123)
      "1.23 Trillion"

      iex> Number.as_human(1234567890123456)
      "1.23 Quadrillion"

      iex> Number.as_human(1234567890123456789)
      "1230 Quadrillion"

      iex> Number.as_human(489939, precision: 2)
      "490 Thousand"

      iex> Number.as_human(489939, precision: 4)
      "489.9 Thousand"

      iex> Number.as_human(1234567, precision: 4, significant: false)
      "1.2346 Million"

      iex> Number.as_human(1234567, precision: 1, separator: ",", significant: false)
      "1,2 Million"

      iex> Number.as_human(500000000, precision: 5)
      "500 Million"

      iex> Number.as_human(12345012345, significant: false)
      "12.345 Billion"

      iex> Number.as_human!("abc")
      ** (ArithmeticError) bad argument in arithmetic expression
  """

  def as_human(...) do

  [...]

  @doc ~S"""
  Throwing version of `as_human`, raises if the input is not a valid number.
  """
  def as_human!(...) do

  [...]
```

**Comment by druzn3k** [comment]
corrected errors

**Comment by duijf** [comment]
:+1:

File: lib/crutches/format/number.ex [hunk 1]
```diff
@@ -418,4 +418,260 @@ defmodule Crutches.Format.Number do
...
+      format: "%n %u"
+    ]
+  ]
+
+  def as_human(number, opts \\ [])
+  def as_human(number, opts) when is_number(number) do
+    opts = Option.combine!(opts, @as_human)
+    {exp, unit, sign} = closest_size_and_sign(number)
+
+    precision = opts[:precision]
```
Comment (CyrusNuevoDia) [inline lib/crutches/format/number.ex:553]:
``` elixir
precision = Math.max(opts[:precision], 0)
```

:shipit:

File: lib/crutches/format/number.ex [hunk 2]
```diff
@@ -418,4 +418,260 @@ defmodule Crutches.Format.Number do
...
+      tenth_exp < -1 && tenth_exp >= -2  -> {-2, :centi, sign}
+      tenth_exp < -2 && tenth_exp >= -3  -> {-3, :milli, sign}
+      tenth_exp < -3 && tenth_exp >= -6  -> {-6, :micro, sign}
+      tenth_exp < -6 && tenth_exp >= -9  -> {-9, :nano, sign}
+      tenth_exp < -9 && tenth_exp >= -12 -> {-12, :pico, sign}
+      tenth_exp < -12 -> {-15, :femto, sign}
+    end
+  end
+
+  defp number_sign(number) when is_number(number) do
```
Comment (CyrusNuevoDia) [inline lib/crutches/format/number.ex:602]: I dunno, I like function heads better. What do you think?

``` elixir
defp number_sign(number) when is_number(number) and number < 0, do: -1
defp number_sign(number) when is_number(number), do: 1
```

File: lib/crutches/format/number.ex [hunk 3]
```diff
@@ -418,4 +418,260 @@ defmodule Crutches.Format.Number do
...
+    cond do
+      number >= 0 ->  1
+      true -> -1
+    end
+  end
+
+  defp rounded_or_significant(number, significant, precision) do
```
Comment (CyrusNuevoDia) [inline lib/crutches/format/number.ex:609]: What do you think of function heads as a replacement for conditionals?

``` elixir
defp rounded_or_significant(number, true, precision), do: make_significant(number, precision)
defp rounded_or_significant(number, false, precision) when precision <= 0 do
  number |> Float.round(precision) |> trunc |> Integer.to_string
end
defp rounded_or_significant(number, false, precision) do
  number |> Float.round(precision) |> :io_lib.format("~.#{precision}f", [number]) |> List.to_string
end
```

Also, now that I've implemented `Number.Format.as_rounded`, should this function use that?

**Pull request merged**